supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/caching/CacheActor.scala

```diff
@@ -0,0 +1,62 @@
+package cromwell.backend.impl.htcondor.caching
+
+import akka.actor.{Actor, ActorLogging}
+import cromwell.backend.BackendJobExecutionActor.SucceededResponse
+import cromwell.backend.impl.htcondor.caching.CacheActor._
+import cromwell.backend.impl.htcondor.caching.exception.{CachedResultAlreadyExistException, CachedResultNotFoundException}
+import cromwell.backend.impl.htcondor.caching.model.CachedExecutionResult
+
+object CacheActor {
+
+  trait CacheActorCommand
+  case class ReadExecutionResult(hash: String) extends CacheActorCommand
+  case class StoreExecutionResult(hash: String, succeededResponse: SucceededResponse) extends CacheActorCommand
+
+  trait CacheActorResponse
+  case class ExecutionResultFound(succeededResponse: SucceededResponse) extends CacheActorResponse
+  case object ExecutionResultNotFound extends CacheActorResponse
+  case class ExecutionResultStored(hash: String) extends CacheActorResponse
+  case object ExecutionResultAlreadyExist extends CacheActorResponse
+
+}
+
+trait CacheActor extends Actor with ActorLogging {
+  def tag: String = "[CacheActor]"
+  def cacheEnabled: Boolean = false
+  def forceRewrite: Boolean = false
+
+  override def receive: Receive = {
+    case ReadExecutionResult(hash) =>
+      try {
+        if (cacheEnabled) {
```

Comment:
I think this cacheEnabled check should be brought up the chain. This actor should only be created / used if cache is enabled. So it can maintain the invariant that if it's created and receives messages, that means that cache must be enabled.